### PR TITLE
refactor(docker): Incrementally migrate keys on each scan result

### PR DESCRIPTION
**Pull request by robzienert**
It occurred to me while running the migration that if network traffic interrupts the scan process, we've done a bunch of work without any actual output. This refactor is more resilient for Spinnaker deployments that are huge.

Changes:

1. Scan operations now retry on errors (wow!)
1. Each scan operation's results are migrated immediately, rather than being aggregated and processed in a second phase.

### CODE REVIEW

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerRegistryCacheV2KeysMigration.java [hunk 1]
```diff
@@ -131,48 +125,75 @@ private void migrateBatch(List<DockerRegistryV1Key> oldKeys) {
...
-        CursorBasedResult<String> result;
+        CursorBasedResult<String> result = null;
         do {
-            result = dyno.dyno_scan(pattern);
-            keys.addAll(oldKeysCallback.apply(result.getResult()));
-        } while (!result.isComplete());
+            try {
+                result = dyno.dyno_scan(pattern);
+                numMigrated += oldKeysCallback.apply(result.getResult());
+            } catch (DynoException e) {
```
Comment (emjburns) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerRegistryCacheV2KeysMigration.java]: Should we keep retrying forever? Are there some exceptions that mean we don't want to retry?

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerRegistryCacheV2KeysMigration.java [hunk 2]
```diff
@@ -131,48 +125,75 @@ private void migrateBatch(List<DockerRegistryV1Key> oldKeys) {
...
+        ScanResult<String> result = null;
         do {
-            result = jedis.scan(cursor, params);
-            keys.addAll(oldKeysCallback.apply(result.getResult()));
-        } while (!result.getStringCursor().equals("0"));
+            try {
+                result = jedis.scan(cursor, params);
+                numMigrated += oldKeysCallback.apply(result.getResult());
+            } catch (JedisException e) {
+                log.error("Failed migrating v1 key batch, retrying", e);
```
Comment (emjburns) [inline igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerRegistryCacheV2KeysMigration.java]: Same comment, do we want to retry forever ever second?

File: igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerRegistryCacheV2KeysMigration.java (hunk 1 above)
Comment (robzienert) [reply igor-web/src/main/groovy/com/netflix/spinnaker/igor/docker/DockerRegistryCacheV2KeysMigration.java]: Good point. I've added a failure counter to abort if there's an excessive amount of consecutive failures.

**Comment by emjburns** [comment]
LTGM

**Pull request merged**